invenio_app_ils/ui/src/invenio_app_ils/common/api/items/item.js

```diff
@@ -4,11 +4,24 @@ import { prepareSumQuery } from '../utils';
 
 const itemURL = '/items/';
 
-const get = itemPid => {
-  return http.get(`${itemURL}${itemPid}`).then(response => {
-    response.data = serializer.fromJSON(response.data);
-    return response;
-  });
+const get = async itemPid => {
+  const response = await http.get(`${itemURL}${itemPid}`);
+  response.data = serializer.fromJSON(response.data);
+  return response;
+};
+
+const del = async itemPid => {
+  const response = await http.delete(`${itemURL}${itemPid}`);
+  return response;
+};
+
+const list = async query => {
+  const response = await http.get(`${itemURL}?q=${query}`);
```

Comment:
no `try/catch`?

Reply:
I think we need to implement a common mechanism for all these so not to bombard with `try/catch` everywhere...

Reply:
@ntarocco @topless I think we should think something similar to this https://reactjs.org/blog/2017/07/26/error-handling-in-react-16.html. Promises that are used directly in the components probably need to be wrapped by an `ErrorBoundary` component and use try/catch in the redux actions as there we can dispatch diffenent actions. WDYT? I can create a quick prototype for that if you want.